modules/telemetry/tests/metric_record_tests.cpp

```diff
@@ -0,0 +1,163 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+
+#include <atomic>
+#include <cstddef>
+#include <cstdint>
+#include <memory>
+#include <random>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "hephaestus/random/random_container.h"
+#include "hephaestus/random/random_generator.h"
+#include "hephaestus/random/random_type.h"
+#include "hephaestus/serdes/json.h"
+#include "hephaestus/telemetry/metric_record.h"
+#include "hephaestus/telemetry/metric_sink.h"
+
```

Comment:
nlohmann as explicit dependency is missing right? shouldn't this be catched by the compiler you did add flags for this right?

Reply:
ahh perhaps it does not work since you only use the macros here....

Reply:
good point, I think the clang-tidy check is not 100% accurate